VideoWeb/VideoWeb/ClientApp/src/app/on-the-day/host-hearing-list/judge-hearing-list/judge-hearing-list.component.ts

```diff
@@ -0,0 +1,70 @@
+import { Component, OnDestroy, OnInit } from '@angular/core';
+import { FormGroup } from '@angular/forms';
+import { Router } from '@angular/router';
+import { Subscription } from 'rxjs';
+import { ProfileService } from 'src/app/services/api/profile.service';
+import { VideoWebService } from 'src/app/services/api/video-web.service';
+import { ConferenceForHostResponse, LoggedParticipantResponse, UserProfileResponse } from 'src/app/services/clients/api-client';
+import { ErrorService } from 'src/app/services/error.service';
+import { EventsService } from 'src/app/services/events.service';
+import { HearingVenueFlagsService } from 'src/app/services/hearing-venue-flags.service';
+import { Logger } from 'src/app/services/logging/logger-base';
+import { vhContactDetails } from 'src/app/shared/contact-information';
+import { ScreenHelper } from 'src/app/shared/screen-helper';
+import { HostHearingListComponent } from 'src/app/on-the-day/host-hearing-list/host-hearing-list.component-base';
+
+@Component({
+    selector: 'app-judge-hearing-list',
+    templateUrl: '../host-hearing-list.component.html',
+    styleUrls: ['../host-hearing-list.component.scss']
+})
+export class JudgeHearingListComponent extends HostHearingListComponent implements OnInit, OnDestroy {
+    contact = {
+        phone: vhContactDetails.phone
+    };
+
+    conferences: ConferenceForHostResponse[];
+    conferencesSubscription = new Subscription();
+    hearingListForm: FormGroup;
+    loadingData: boolean;
+    interval: any;
+    today = new Date();
+    profile: UserProfileResponse;
+    loggedUser: LoggedParticipantResponse;
+    eventHubSubscriptions: Subscription = new Subscription();
+
+    constructor(
+        private errorService: ErrorService,
+        protected videoWebService: VideoWebService,
+        protected router: Router,
+        protected profileService: ProfileService,
+        protected logger: Logger,
+        protected eventsService: EventsService,
+        protected screenHelper: ScreenHelper,
+        protected hearingVenueFlagsService: HearingVenueFlagsService
+    ) {
+        super(videoWebService, router, profileService, logger, eventsService, screenHelper, hearingVenueFlagsService);
+    }
+
+    retrieveHearingsForUser() {
+        this.logger.debug('[JudgeHearingList] - Updating hearing list');
+        this.conferencesSubscription.add(
```

Comment:
Do these get unsubscribed anywhere?

Reply:
this gets unsubscribed in the base component, does the ngOnDestroy of the base component get called by the child components?

Reply:
I will check online, but normally that should behave that way

Reply:
It does. Unless you override it in which case you would need to call super.ngOnDestroy but you don't so that's fine.

Reply:
so it does get called, however this will get overriden if we in the future create an ngOnDestroy in the child component, we will need to ensure that super.ngOnDestroy is called first if we ever have to implement OnDestroy